cosmosdb/cosmosdb_endpoint.bal

```diff
@@ -0,0 +1,725 @@
+    string resourceLink = DB_PATH + SLASH + databaseId;
+    string requestPath = SLASH + resourceLink;
+    http:Request request = new;
+    var requestHeaders = costructRequestHeaders(request, "delete", DB_PATH, resourceLink, self.masterKey,
+        JSON_CONTENT_TYPE, requestOptions = requestOptions);
+    if (requestHeaders is error) {
+        error err = error(COSMOS_DB_ERROR_CODE,
+        { message: "Error occurred while encoding parameters when constructing request headers" });
+        return err;
+    } else {
+        var httpResponse = self.documentClient->delete(requestPath, request);
+        if (httpResponse is http:Response) {
+            int statusCode = httpResponse.statusCode;
+            var jsonResponse = httpResponse.getJsonPayload();
+            if (jsonResponse is json) {
+                if (statusCode == http:NO_CONTENT_204) {
+                    ResourceResponse resourceResponse = extractResponseHeaders(httpResponse);
+                    return resourceResponse;
+                } else {
+                    return setResponseError(jsonResponse);
+                }
+            } else {
+                error err = error(COSMOS_DB_ERROR_CODE,
+                { message: "Error occurred while accessing the JSON payload of the response" });
+                return err;
+            }
+        } else {
+            error err = error(COSMOS_DB_ERROR_CODE, { message: "Error occurred while invoking the REST API" });
+            return err;
+        }
+    }
+}
+
+remote function Client.createCollection(string databaseId, string collectionId,
+                                        CollectionCreateOptions? collectionCreateOptions = ()) returns
+                                                                                               CollectionResponse|error
+{
+    string resourceLink = DB_PATH + SLASH + databaseId;
+    string requestPath = SLASH + resourceLink + SLASH + COLL_PATH;
+    http:Request request = new;
+    json collectionJSONPayload = { "id": collectionId };
+    request.setJsonPayload(collectionJSONPayload);
+    var requestHeaders = costructRequestHeaders(request, "post", COLL_PATH, resourceLink, self.masterKey,
+        JSON_CONTENT_TYPE, requestOptions = collectionCreateOptions.requestOptions);
+    if (requestHeaders is error) {
+        error err = error(COSMOS_DB_ERROR_CODE,
+        { message: "Error occurred while encoding parameters when constructing request headers" });
+        return err;
+    } else {
+        //Set collection specific headers.
+        if (collectionCreateOptions != ()) {
+            if (collectionCreateOptions.offerThroughput != ()) {
+                request.setHeader("x-ms-offer-throughput", <string>collectionCreateOptions.offerThroughput);
+            }
+            if (collectionCreateOptions.offerType != ()) {
+                request.setHeader("x-ms-offer-type", <string>collectionCreateOptions.offerType);
```

Comment:
Shall we follow the same convention for all the headers? Some are defined as constants and others not. Check all the places.